be/extmap.c

```diff
@@ -418,6 +418,19 @@ M0_INTERNAL void m0_be_emap_create(struct m0_be_emap   *map,
 	m0_be_op_done(op);
 }
 
+M0_INTERNAL void m0_be_emap_truncate(struct m0_be_emap *map,
+				     struct m0_be_tx   *tx,
+				     struct m0_be_op   *op,
+				     m0_bcount_t       *limit)
+{
+	struct m0_btree_op b_op = {};
+	m0_be_op_active(op);
+	op->bo_u.u_emap.e_rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op,
+			       m0_btree_truncate(map->em_mapping, *limit,
+						 tx, &b_op));
+	m0_be_op_done(op);
```

Comment:
Does m0_btree_truncate() return a different return code if it has run out of credits while there are more records to be deleted?

Reply:
I need to check with Nikhil. Earlier, I was assuming it will return a nonzero return code if has run out of credits and still contains some records in the tree.

Reply:
m0_btree_truncate() will not return any error code if it runs out of credits. The m0_btree_truncate is implemented in line with old btree truncate function, where it will delete the records/nodes as per the credit passed. The caller function will check if tree is empty using m0_btree_is_empty() . If not empty then the caller will call m0_btree_truncate() again.

Reply:
Made changes to call m0_btree_is_empty instead of relying on a return code to find if all records are deleted by truncate.